snap/snapcraft.yaml

```diff
@@ -6,30 +6,56 @@ description: |
   not yet supported by the version of fwupd distributed with the OS.
 
 grade: stable
-confinement: classic
+confinement: strict
 base: core22
 
 architectures:
-  - amd64
+  - build-on: amd64
+
+slots:
+  fwupd:
+    interface: fwupd
+  fwupd-dbus:
+    interface: dbus
+    bus: system
+    name: org.freedesktop.fwupd
+
+plugs:
+  fwupdmgr:
+    interface: fwupd
+  polkit:
+    interface: polkit
+    action-prefix: org.freedesktop.fwupd
 
 apps:
   dfu-tool:
     command: dfu-tool.wrapper
+    plugs: [fwupdmgr, network]
   dbxtool:
     command: dbxtool.wrapper
+    plugs: [fwupdmgr, network]
   fwupdtool:
     command: fwupdtool.wrapper
+    plugs: [bluez, udisks2, modem-manager, upower-observe, network]
+    slots: [fwupd]
     completer:
       share/bash-completion/completions/fwupdtool
   fwupd:
     command: fwupd.wrapper
-    daemon: simple
+    daemon: dbus
+    slots: [fwupd]
+    plugs: [udisks2, modem-manager, upower-observe, polkit, network]
```

Comment:
don't you need bluez here too?

Reply:
Yes, and that's an internal network not the internet. Not sure if snap can confine to internal adapters or not.

Reply:
Yes. I think we need it too. I will test that again and verify there are errors.

Reply:
The snapd interfaces has no netfilter backend. So I do not think we could easily make an interface for specific networks.

However, we could add more restricted network access in the `fwupd` slot. For example, allow just tcp, and not allow resolve.

Reply:
@hughsie do the internal networks ever use DNS that you've seen though?  It's probably better not to back ourselves into a corner if they expect to use mDNS or something like `redfish.local`

Reply:
I've only ever seen ipv4 addresses, but it's not inconceivable to see a hostname there.

Reply:
I'd say we should avoid locking this down any further for now.